src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs

```diff
@@ -43,7 +43,12 @@ public interface IUntypedObject
         // It is validated to be in a culture invariant standard format number, possibly with dot decimal separator and "E" exponent.
         // This method need not be implemetned if ExteranlType.UntypedNumber is not used.
         // GetDouble and GetDecimal can be called on an ExternalType.UntypedNumber.
-        string GetUntypedNumber();
+        string GetUntypedNumber();
+
+        /// <summary>
+        /// Returns the list of available properties if the current instance is an object.
+        /// </summary>
+        IEnumerable<string> GetProperties();
```

Comment:
Suggestion: `GetPropertyNames` - TryGetProperty gets the property value; this is returning the property names, not their values.